ansible_base/oauth2_provider/views/permissions.py

```diff
@@ -0,0 +1,37 @@
+from django.conf import settings
+from rest_framework.permissions import SAFE_METHODS, BasePermission
+
+
+class OAuth2TokenPermission(BasePermission):
+    # An app token is a token that has an application attached to it
+    # A personal access token (PAT) is a token with no application attached to it
+    # With that in mind:
+    # - An app token can be read, changed, or deleted if:
+    #   - I am the superuser
+    #   - I am the admin of the organization of the application of the token
+    #   - I am the user of the token
+    # - An app token can be created if:
+    #   - I have read access to the application (currently this means: I am the superuser)
+    # - A PAT can be read, changed, or deleted if:
+    #   - I am the superuser
+    #   - I am the user of the token
+    # - A PAT can be created if:
+    #   - I am a user
+
+    def has_permission(self, request, view):
+        # Handle PAT and app token creation separately
+        if request.method == 'POST':
+            if request.data.get('application'):
+                # TODO: Change this once ansible/django-ansible-base#424 is fixed
+                return request.user.is_superuser
+        return request.user.is_authenticated
+
+    def has_object_permission(self, request, view, obj):
+        if request.method in SAFE_METHODS and getattr(request.user, 'is_system_auditor', False):
```

Comment:
I really don't want to keep a is_system_auditor check long term, but I guess that should be replaced as soon as we're able to add the model to the registry.